src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -904,76 +905,154 @@ namespace
     {
         if (rend_params.envMap != nullptr)
         {
-            const size_t TextureWidth = 512;
-            const size_t TextureHeight = 512;
-
-            // Render the environment map into a Max bitmap.
-            BitmapInfo bi;
-            bi.SetWidth(TextureWidth);
-            bi.SetHeight(TextureHeight);
-            bi.SetType(BMM_FLOAT_RGBA_32);
-            Bitmap* envmap_bitmap = TheManager->Create(&bi);
-            rend_params.envMap->RenderBitmap(time, envmap_bitmap, 1.0f, TRUE);
-
-            // Build an appleseed image from the Max bitmap.
-            asf::auto_release_ptr<asf::Image> envmap_image(
-                new asf::Image(
-                    TextureWidth, TextureHeight,    // image dimensions
-                    TextureWidth, TextureHeight,    // tile dimensions
-                    4,
-                    asf::PixelFormatFloat));
-            for (size_t y = 0; y < TextureHeight; ++y)
+            std::string env_edf_name("environment_edf");
+            std::string env_shader_name("environment_shader");
+            std::string env_tex_name("environment_map");
+            std::string env_tex_instance_name("environment_map_inst");
+
+            //insert textures
+            //in case of bitmap env map
+            if (rend_params.envMap->IsSubClassOf(Class_ID(BMTEX_CLASS_ID, 0)))
             {
-                for (size_t x = 0; x < TextureWidth; ++x)
+                auto bitmap_envmap = static_cast<BitmapTex*>(rend_params.envMap);
+                if (bitmap_envmap)
                 {
-                    BMM_Color_fl c;
-                    envmap_bitmap->GetLinearPixels(
-                        static_cast<int>(x),
-                        static_cast<int>(y),
-                        1,
-                        &c);
-                    envmap_image->set_pixel(x, y, c);
+                    env_tex_instance_name = insert_texture_and_instance(scene, bitmap_envmap);
                 }
             }
+            //in case of unknown class
+            else if (!rend_params.envMap->IsSubClassOf(AppleseedEnvMap::get_class_id()))
+            {
+                //proceed with rendering env map and applying it to background shader
```

Comment:
Full line comments should start with a capital and end with a period. There should also be a space after `//`:

    // Proceed with rendering env map and apply it to background shader.